src/Test.UnitTests.Sarif/Baseline2/MatchedResultsTests.cs

```diff
@@ -0,0 +1,291 @@
+                    {
+                        Provenance = new ResultProvenance
+                        {
+                            FirstDetectionTimeUtc = firstDetectionTime
+                        }
+                    },
+                    CurrentRun = new Run(),
+                    ExpectedBaselineState = BaselineState.New,
+                    ExpectedFirstDetectionTime = firstDetectionTime
+                },
+                new FirstDetectionTimeTestCase
+                {
+                    Name = "New, current run end time only",
+                    PreviousResult = null,
+                    PreviousRun = new Run(),
+                    CurrentResult = new Result(),
+                    CurrentRun = new Run
+                    {
+                        Invocations = new List<Invocation>
+                        {
+                            new Invocation
+                            {
+                                EndTimeUtc = currentRunEndTime
+                            }
+                        }
+                    },
+                    ExpectedBaselineState = BaselineState.New,
+                    ExpectedFirstDetectionTime = currentRunEndTime
+                },
+                new FirstDetectionTimeTestCase
+                {
+                    Name = "New, current run start time only",
+                    PreviousResult = null,
+                    PreviousRun = new Run(),
+                    CurrentResult = new Result(),
+                    CurrentRun = new Run
+                    {
+                        Invocations = new List<Invocation>
+                        {
+                            new Invocation
+                            {
+                                StartTimeUtc = currentRunStartTime
+                            }
+                        }
+                    },
+                    ExpectedBaselineState = BaselineState.New,
+                    ExpectedFirstDetectionTime = currentRunStartTime
+                },
+                new FirstDetectionTimeTestCase // TODO TEST CASE SHOWING CURRENT RUN WINS FOR NEW AND PREV RUN WINS FOR EXISTING
+                {
+                    Name = "New, current start and end times",
+                    PreviousResult = null,
+                    PreviousRun = new Run(),
+                    CurrentResult = new Result(),
+                    CurrentRun = new Run
+                    {
+                        Invocations = new List<Invocation>
+                        {
+                            new Invocation
+                            {
+                                StartTimeUtc = currentRunStartTime,
+                                EndTimeUtc = currentRunEndTime
+                            }
+                        }
+                    },
+                    ExpectedBaselineState = BaselineState.New,
+                    ExpectedFirstDetectionTime = currentRunEndTime
```

Comment:
If/When you add logic to set LastDetectionTime, FirstDetectionRunGuid, and LastDetectionRunGuid, check for those, too.

Reply:
Sure, this will be the natural place to add those. WCP doesn't require any of those.

---
In reply to: [372135318](https://github.com/microsoft/sarif-sdk/pull/1739#discussion_r372135318) [](ancestors = 372135318)